amr-wind/mesh_mapping_models/ChannelFlowScaling.cpp

```diff
@@ -0,0 +1,334 @@
+                scale_fac_xf(i, j, k, 1) =
+                    (do_map[1] && in_domain) ? fac_y : 1.0;
+                scale_fac_xf(i, j, k, 2) =
+                    (do_map[2] && in_domain) ? fac_z : 1.0;
+            });
+    }
+
+    for (amrex::MFIter mfi(m_mesh_scale_fac_yf(lev)); mfi.isValid(); ++mfi) {
+
+        const auto& bx = mfi.growntilebox();
+        amrex::Array4<amrex::Real> const& scale_fac_yf =
+            m_mesh_scale_fac_yf(lev).array(mfi);
+        amrex::ParallelFor(
+            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];
+                amrex::Real y = prob_lo[1] + j * dx[1];
+                amrex::Real z = prob_lo[2] + (k + 0.5) * dx[2];
+
+                amrex::Real fac_x = eval_fac(x, beta[0], prob_lo[0], len[0]);
+                amrex::Real fac_y = eval_fac(y, beta[1], prob_lo[1], len[1]);
+                amrex::Real fac_z = eval_fac(z, beta[2], prob_lo[2], len[2]);
+
+                bool in_domain =
+                    ((x > prob_lo[0]) && (x < prob_hi[0]) &&
+                     (y >= prob_lo[1] - eps) && (y <= prob_hi[1] + eps) &&
+                     (z > prob_lo[2]) && (z < prob_hi[2]));
+
+                scale_fac_yf(i, j, k, 0) =
+                    (do_map[0] && in_domain) ? fac_x : 1.0;
+                scale_fac_yf(i, j, k, 1) =
+                    (do_map[1] && in_domain) ? fac_y : 1.0;
+                scale_fac_yf(i, j, k, 2) =
+                    (do_map[2] && in_domain) ? fac_z : 1.0;
+            });
+    }
+
+    for (amrex::MFIter mfi(m_mesh_scale_fac_zf(lev)); mfi.isValid(); ++mfi) {
+
+        const auto& bx = mfi.growntilebox();
+        amrex::Array4<amrex::Real> const& scale_fac_zf =
+            m_mesh_scale_fac_zf(lev).array(mfi);
+        amrex::ParallelFor(
+            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];
+                amrex::Real y = prob_lo[1] + (j + 0.5) * dx[1];
+                amrex::Real z = prob_lo[2] + k * dx[2];
+
+                amrex::Real fac_x = eval_fac(x, beta[0], prob_lo[0], len[0]);
+                amrex::Real fac_y = eval_fac(y, beta[1], prob_lo[1], len[1]);
+                amrex::Real fac_z = eval_fac(z, beta[2], prob_lo[2], len[2]);
+
+                bool in_domain =
+                    ((x > prob_lo[0]) && (x < prob_hi[0]) && (y > prob_lo[1]) &&
+                     (y < prob_hi[1]) && (z >= prob_lo[2] - eps) &&
+                     (z <= prob_hi[2] + eps));
+
+                scale_fac_zf(i, j, k, 0) =
+                    (do_map[0] && in_domain) ? fac_x : 1.0;
+                scale_fac_zf(i, j, k, 1) =
+                    (do_map[1] && in_domain) ? fac_y : 1.0;
+                scale_fac_zf(i, j, k, 2) =
+                    (do_map[2] && in_domain) ? fac_z : 1.0;
+            });
+    }
+
+    // TODO: Call fill patch operators
```

Comment:
Relevant comment?

Reply:
yes, just leaving it in there because it still isn't clear to use if fill patch operators are something that we would need to call here.